src/Dialect/ONNX/ONNXDimAnalysis.cpp

```diff
@@ -592,6 +592,37 @@ void DimAnalysis::visitDim(
     return;
   }
 
+  // TileOp
+  if (auto tileOp = dyn_cast<ONNXTileOp>(op)) {
+    if (!areDimsFromConcat(tileOp.getRepeats()))
```

Comment:
Early `return` may block general case for tileOp.

We should call `exploreSameInputDims(dim, op, sameDims);` at the beginning to capture all general cases for tileOp, then go to special case. See MatMul as an example.

Another way is only use `return` inside the `if` of the special case, to let other cases go through (See ExpandOp as an example):
```
if (auto tileOp = dyn_cast<ONNXTileOp>(op)) {
   if (areDimsFromConcat(tileOp.getRepeats()) && inputShape[dimIndex] == 1) {
      do something;
     return;
  }
}
```

Reply:
I recommend the latter way.

Reply:
@tungld I used the latter way. Thanks for your comment!